test/TestProjects/MgmtResourceName/Generated/MemoryResourceCollection.cs

```diff
@@ -0,0 +1,353 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Core;
+using Azure.ResourceManager.Resources;
+
+namespace MgmtResourceName
+{
+    /// <summary> A class representing collection of MemoryResource and their operations over its parent. </summary>
+    public partial class MemoryResourceCollection : ArmCollection, IEnumerable<MemoryResource>, IAsyncEnumerable<MemoryResource>
```

Comment:
Hey @m-nash in this test project, we have the `request-path-to-resource-name` configuration which assigned the resource name of this to `MemoryResource` (which ends with `Resource`) and as you can see here, the collection keeps the `Resource` suffix from the config. Same thing needs to be done on `GenericResource` and we can have `GenericResourceCollection`.